crates/net/discv4/src/config.rs

```diff
@@ -8,14 +8,17 @@ use reth_net_common::ban_list::BanList;
 use reth_net_nat::{NatResolver, ResolveNatInterval};
 use reth_primitives::NodeRecord;
 use reth_rlp::Encodable;
-use secp256k1::serde::{Deserialize, Serialize};
 use std::{
     collections::{HashMap, HashSet},
     time::Duration,
 };
 
+#[cfg(feature = "serde")]
+use secp256k1::serde::{Deserialize, Serialize};
```

Comment:
oh, let's use the right `serde::{Deserialize, Serialize}` traits here instead of using the reexports.